core/src/execution/mod.rs

```diff
@@ -337,6 +337,7 @@ impl<N: NetworkSpec, R: ExecutionRpc<N>> ExecutionClient<N, R> {
             .collect::<Result<HashSet<_>, _>>()?;
 
         // Collect all (proven) tx receipts as a map of tx hash to receipt
+        // TODO: use get_block_receipts instead to reduce the number of RPC calls?
```

Comment:
Can significantly optimize `verify_logs` too. Will create a separate issue for it once this PR is merged.

Reply:
Issue: https://github.com/a16z/helios/issues/438